lib/ansible/module_utils/shell.py

```diff
@@ -256,22 +256,31 @@ def to_command(self, obj):
             return cmdobj
 
         elif not isinstance(obj, dict):
-            transform = ComplexDict(dict(
-                command=dict(key=True),
-                prompt=dict(),
-                answer=dict(),
-                sendonly=dict(default=False)
-            ))
+            if module:
+                transform = ComplexDict(dict(
+                    command=dict(key=True),
+                    prompt=dict(),
+                    answer=dict(),
+                    sendonly=dict(default=False)
+                ), module)
+            else:
+                transform = ComplexDict(dict(
```

Comment:
No, this was just to get it working and not touch anything for other modules. But I agree that this and other things needs to be changed.